DQM/Physics/src/TopSingleLeptonDQM_miniAOD.cc

```diff
@@ -0,0 +1,789 @@
+          // restrict to the leading electron
+          fill("elecPt_", elec->pt());
+          fill("elecEta_", elec->eta());
+          fill("elecRelIso_", el_pfRelIso);
+          fill("elecChHadIso_", el_ChHadIso);
+          fill("elecNeHadIso_", el_NeHadIso);
+          fill("elecPhIso_", el_PhIso);
+        }
+
+        ++eMult;
+        if (!elecIso_ || (*elecIso_)(*elec)) {
+          isoElecs.push_back(&(*elec));
+          ++eMultIso;
+        }
+      }
+    }
+  }
+  fill("elecMult_", eMult);
+  fill("elecMultIso_", eMultIso);
+
+  /*
+  ------------------------------------------------------------
+
+  Muon Monitoring
+
+  ------------------------------------------------------------
+  */
+
+  // fill monitoring plots for muons
+  unsigned int mMult = 0, mMultIso = 0;
+
+  edm::Handle<edm::View<pat::Muon> > muons;
+  edm::View<pat::Muon>::const_iterator muonit;
+
+  if (!event.getByToken(muons_, muons)) return;
+
+
+
+  for (edm::View<pat::Muon>::const_iterator muon = muons->begin();
+       muon != muons->end(); ++muon) {
+
+
+    // restrict to globalMuons
+    if (muon->isGlobalMuon()) {
+      fill("muonDelZ_", muon->innerTrack()->vz());  // CB using inner track!
+      fill("muonDelXY_", muon->innerTrack()->vx(), muon->innerTrack()->vy());
+
+
+
+      // apply preselection
+      if (!muonSelect_ || (*muonSelect_)(*muon)) {
+ 
+
+        double chHadPt = muon->pfIsolationR04().sumChargedHadronPt;
+        double neHadEt = muon->pfIsolationR04().sumNeutralHadronEt;
+        double phoEt = muon->pfIsolationR04().sumPhotonEt;
+
+        double pfRelIso =
+            (chHadPt +
+             max(0., neHadEt + phoEt - 0.5 * muon->pfIsolationR04().sumPUPt)) /
+            muon->pt();  // CB dBeta corrected iso!
+
+        if (mMult == 0) {
+          // restrict to leading muon
+          fill("muonPt_", muon->pt());
+          fill("muonEta_", muon->eta());
+
+          fill("muonRelIso_", pfRelIso);
+
+          fill("muonChHadIso_", chHadPt);
+          fill("muonNeHadIso_", neHadEt);
+          fill("muonPhIso_", phoEt);
+        }
+        ++mMult;
+        if (!muonIso_ || (*muonIso_)(*muon)) ++mMultIso;
+      }
+    }
+  }
+  fill("muonMult_", mMult);
+  fill("muonMultIso_", mMultIso);
+
+  /*
+  ------------------------------------------------------------
+
+  Jet Monitoring
+
+  ------------------------------------------------------------
+  */
+
+
+  // loop jet collection
+  std::vector<pat::Jet> correctedJets;
+  std::vector<double> JetTagValues;
+  unsigned int mult = 0, multBEff = 0, multBPur = 0, multBVtx = 0, multCSV = 0;
+
+  edm::Handle<edm::View<pat::Jet> > jets;
+  if (!event.getByToken(jets_, jets)) {
+    return;
+  }
+
+  for (edm::View<pat::Jet>::const_iterator jet = jets->begin();
+       jet != jets->end(); ++jet) {
+    // check jetID for calo jets
+    unsigned int idx = jet - jets->begin();
+
+      pat::Jet sel = *jet;
+
+      StringCutObjectSelector<pat::Jet> jetSelect(jetSelect_);
```

Comment:
@pargali - please make this StringCutObjectSelect<pat::Jet> once per job not once per jet. Its quite expensive